nexus/src/external_api/shared.rs

```diff
@@ -0,0 +1,221 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Types that are used as both views and params
```

Comment:
I agree that the current split is confusing.  I understand the current guidance to be:

* if it's an input param, it goes in "params"
* if it's an output, it goes in "views"
* if it's both, it goes in "shared"
* there's also a bunch of stuff in "common" that hasn't been put into one of those three places because it's hard

This last point makes it kind of a mess, and it's also possible that this last part is hard because there are other constraints involved that this guidance doesn't accommodate.  Progenitor changes a bunch of this too -- I think part of the original reason for "common" was for stuff that was shared between clients and servers of a given service.

On your specific example:

> We could, for example, have groups of endpoints and types live together.

We could.  Any way I can think to group them sucks a little bit -- including the current approach of putting them in one big file.  (If we group them by resource, one might be sad that the various "create" endpoints (which are very parallel to each other and use some of the same types, like IdentityMetadataCreate) don't live near each other.  Also, it's kind of nice for some small groups of resources to live together (e.g., built-in users and roles, which might go under an "IAM" bucket).  So maybe this is by tag after all?)

Basically, I don't have a strong idea of how these should be better organized.